tests/conftest.py

```
@@ -47,10 +47,21 @@ def start_shiny_server():
         - scope="session" → Server runs ONCE for entire test session
-        - autouse=True → No need to request this fixture in tests
-        - This is the fastest approach for E2E tests
+        - Only runs when explicitly requested (e.g., for E2E tests)
+        - Unit tests marked with @pytest.mark.unit skip this fixture
     """
     
+    # Check if this is a unit test run - skip server startup
+    # This is triggered when running with -m unit or when all tests are unit tests
+    collected_items = getattr(request.session, 'items', [])
+    if collected_items and all(
+        any(mark.name == 'unit' for mark in getattr(item, 'iter_markers', lambda: [])())
```

Ruff (0.14.10)

Prefer `list` over useless lambda

Replace with `lambda` with `list`

(PIE807)